src/main/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStep.java

```diff
@@ -105,6 +111,28 @@ public DescriptorImpl() {
             return true;
         }
 
+        @Override public UninstantiatedDescribable uninstantiate(Step step) throws UnsupportedOperationException {
+            RegistryEndpointStep s = (RegistryEndpointStep) step;
+            Map<String, Object> args = new TreeMap<>();
+            args.put("url", s.registry.getUrl());
+            args.put("credentialsId", s.registry.getCredentialsId());
+            args.put("toolName", s.toolName);
+            args.values().removeAll(Collections.singleton(null));
+            return new UninstantiatedDescribable(args);
+        }
+
+        @Override public Step newInstance(Map<String, Object> arguments) throws Exception {
+            if (arguments.containsKey("url") || arguments.containsKey("credentialsId")) {
```

Comment:
Direct mutation of input Map: danger, danger will robinson! 

Please create copy if you're going to mutate something (just in case something else is done with the args) -- don't remember off the top of my head but there's a real risk this might break something like ArgumentsActionImpl.

Reply:
It's not a hard-blocker for a known-and-significant error, but it's *dangerous* to be doing this here.

Reply:
(resolved)